app/lib/document_navigation.rb

```diff
@@ -1,107 +1,81 @@
+    Documents::Form1099csController,
+    Documents::Form1099divsController,
+    Documents::Form1099intsController,
+    Documents::Form1099ksController,
+    Documents::Form1099miscsController,
+    Documents::Form1099rsController,
+    Documents::Form1099ssController,
+    Documents::Form1099sasController,
+    Documents::Form1099gsController,
+    Documents::Form5498sasController,
+    Documents::IraStatementsController,
+    Documents::PropertyTaxStatementsController,
+    Documents::Rrb1099sController,
+    Documents::Ssa1099sController,
+    Documents::StudentAccountStatementsController,
+    Documents::CareProviderStatementsController,
+    Documents::W2gsController,
+    Documents::PriorTaxReturnsController,
+    Documents::AdditionalDocumentsController,
+    Documents::RequestedDocumentsController,
+    Documents::RequestedDocumentsLaterController,
     Documents::OverviewController,
     Documents::SelfieInstructionsController,
     Documents::SendRequestedDocumentsController,
     Documents::SendRequestedDocumentsLaterController,
   ].freeze
-  DOCUMENT_TYPES = DOCUMENT_CONTROLLERS.keys.freeze
+
+  DOCUMENT_TYPES = FLOW.map(&:document_type).compact
 
   class << self
     delegate :first, to: :controllers
 
     def controllers
-      DOCUMENT_CONTROLLERS.values
+      FLOW
     end
 
-    def all_controllers
-      BEFORE_CONTROLLERS + controllers + AFTER_CONTROLLERS
+    def first_for_intake(intake)
+      controllers.find { |c| c.show?(intake) }
     end
 
-    def document_type(controller_class)
-      DOCUMENT_CONTROLLERS.key(controller_class)
+    def document_types_for_intake(intake)
+      controllers
+        .find_all { |c| c.show?(intake) }
+        .map(&:document_type)
+        .compact
+    end
+
+    def document_controller_for_type(document_type)
+      controllers.find { |c| c.document_type == document_type }
     end
   end
 
   delegate :controllers, to: :class
-  delegate :all_controllers, to: :class
-  delegate :document_type, to: :class
 
   def initialize(current_controller)
     @current_controller = current_controller
   end
 
-  def next
-    return unless index
-
-    controllers_until_end = all_controllers[(index + 1)..-1]
-    seek(controllers_until_end)
-  end
-
-  def first_for_intake(intake)
-    select(intake).first
-  end
-
-  def select(intake)
-    controllers.select do |controller_class|
-      controller_class.show?(intake)
-    end
-  end
-
-  def types_for_intake(intake)
-    select(intake).map do |controller_class|
-      document_type(controller_class)
-    end
-  end
+  def next_for_intake(intake)
+    current_index = controllers.index(@current_controller.class)
+    return if current_index.nil?
 
-  private
-
-  def index
-    all_controllers.index(@current_controller.class)
-  end
-
-  def seek(list)
-    list.detect do |controller_class|
-      controller_class.show?(@current_controller.current_intake)
-    end
+    controllers[(current_index + 1)..-1].find { |c| c.show?(intake) }
```

Comment:
This is terse but dense! Non-blocking, but I think this would be more readable if broken up into separate statements.